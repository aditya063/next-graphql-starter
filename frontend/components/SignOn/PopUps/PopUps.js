/* eslint-disable react/no-unused-prop-types */
import PropTypes from 'prop-types';
import SignIn from '../SignIn/SignIn';
import SignUp from '../SignUp/SignUp';
import RequestReset from '../RequestReset/RequestReset';
import * as Styled from './PopUps.style';

const PopUps = React.memo(props => {
  const showPopUp = () => {
    switch (props.popUp) {
      case 'signIn':
        return (
          <SignIn
            close={() => props.setPopUp('')}
            requestReset={() => props.setPopUp('requestReset')}
          />
        );

      case 'signUp':
        return <SignUp close={() => props.setPopUp('')} />;

      case 'requestReset':
        return <RequestReset close={() => props.setPopUp('')} />;

      default:
        return null;
    }
  };

  return (
    <Styled.divPopup>
      <Styled.divOuter onClick={() => props.setPopUp('')} />

      <Styled.divInner>{showPopUp()}</Styled.divInner>
    </Styled.divPopup>
  );
});

PopUps.propTypes = {
  popUp: PropTypes.string.isRequired,
  setPopUp: PropTypes.func.isRequired
};

export default PopUps;
