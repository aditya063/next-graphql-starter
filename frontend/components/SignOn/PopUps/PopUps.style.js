import styled from 'styled-components';

export const divPopup = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.5);
`;

export const divOuter = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`;

export const divInner = styled.div`
  position: absolute;
  padding: 10px;
  background-color: white;
`;
