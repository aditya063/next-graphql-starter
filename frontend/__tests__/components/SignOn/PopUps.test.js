import React from 'react';
import { shallow } from 'enzyme';
import PopUps from '../../../components/SignOn/PopUps/PopUps';

describe('PopUps', () => {
  let mockProps;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    wrapper = shallow(<PopUps {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
