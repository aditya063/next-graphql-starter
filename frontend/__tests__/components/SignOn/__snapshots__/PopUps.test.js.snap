// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUps matches snapshot 1`] = `
.c0 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  z-index: 666;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0,0,0,0.5);
}

.c2 {
  z-index: 999;
  padding: 10px;
  background-color: white;
}

<PopUpsstyle__divPopup>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "PopUpsstyle__divPopup-sc-1nrsy50-0",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "position:absolute;top:0;bottom:0;left:0;right:0;display:flex;justify-content:center;align-items:center;",
          ],
        },
        "displayName": "PopUpsstyle__divPopup",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "PopUpsstyle__divPopup-sc-1nrsy50-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    >
      <PopUpsstyle__divOuter
        onClick={[Function]}
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "PopUpsstyle__divOuter-sc-1nrsy50-1",
                "isStatic": true,
                "lastClassName": "c1",
                "rules": Array [
                  "z-index:666;position:absolute;top:0;bottom:0;left:0;right:0;background-color:rgba(0,0,0,0.5);",
                ],
              },
              "displayName": "PopUpsstyle__divOuter",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "PopUpsstyle__divOuter-sc-1nrsy50-1",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          onClick={[Function]}
        >
          <div
            className="c1"
            onClick={[Function]}
          />
        </StyledComponent>
      </PopUpsstyle__divOuter>
      <PopUpsstyle__divInner>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "PopUpsstyle__divInner-sc-1nrsy50-2",
                "isStatic": true,
                "lastClassName": "c2",
                "rules": Array [
                  "z-index:999;padding:10px;background-color:white;",
                ],
              },
              "displayName": "PopUpsstyle__divInner",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "PopUpsstyle__divInner-sc-1nrsy50-2",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c2"
          />
        </StyledComponent>
      </PopUpsstyle__divInner>
    </div>
  </StyledComponent>
</PopUpsstyle__divPopup>
`;
